### PR TITLE
feat: Support Single Tenant authentication through BotFramework-Emulator

### PR DESCRIPTION
**Pull request by JhontSouth**
#minor

## Description
This PR includes a dynamic way to validate the access token _**issuer**_ value depending on the tenant ID used in Emulator.

## Specific Changes
  - Added logic to validate the _**tid**_ and _**issuer**_ values to allow the Single Tenant authentication

## Testing
The following image shows the Emulator working with a bot using Single Tenant authentication.
image

### CODE REVIEW

File: libraries/botframework-connector/src/auth/emulatorValidation.ts [hunk 1]
```diff
@@ -75,6 +75,15 @@ export namespace EmulatorValidation {
             return false;
         }
 
+        if (Array.isArray(ToBotFromBotOrEmulatorTokenValidationParameters.issuer)) {
```
Comment (ceciliaavila) [inline libraries/botframework-connector/src/auth/emulatorValidation.ts:78]: This method is very well documented. We should include some comments explaining what we are evaluating here.

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

File: libraries/botframework-connector/src/auth/emulatorValidation.ts (hunk 1 above)
Comment (sw-joelmut) [reply libraries/botframework-connector/src/auth/emulatorValidation.ts:79]: same here as mentioned in DotNet's pr

**Review by sw-joelmut — CHANGES_REQUESTED** [review]

File: libraries/botframework-connector/src/auth/emulatorValidation.ts (hunk 1 above)
Comment (JhontSouth) [reply libraries/botframework-connector/src/auth/emulatorValidation.ts:79]: @ceciliaavila @sw-joelmut 
feedback applied

**Review by sw-joelmut — APPROVED** [review]

File: libraries/botframework-connector/src/auth/emulatorValidation.ts (hunk 1 above)
Comment (sw-joelmut) [reply libraries/botframework-connector/src/auth/emulatorValidation.ts:79]: I think we could add the `tid` like dotnet, just to maintain parity, even though its not being used here `const tenantId = token?.payload?.tid ?? '';` or maybe we could use it like this `const tenantId = token?.payload?.[AuthenticationConstants.TenantIdClaim] ?? '';`

dotnet =>
image

**Review by ceciliaavila — APPROVED** [review]

File: libraries/botframework-connector/src/auth/emulatorValidation.ts (hunk 1 above)
Comment (JhontSouth) [reply libraries/botframework-connector/src/auth/emulatorValidation.ts:79]: fixed

**Comment by JhontSouth** [comment]
Promoted /4643

**Pull request closed without merging**